ff/serving/metadata/metadata_test.go

```diff
@@ -0,0 +1,1105 @@
+}
+
+func (m *TrainingSetStreamMock) Context() context.Context {
+	return m.ctx
+}
+func (m *TrainingSetStreamMock) Send(resp *pb.TrainingSet) error {
+	m.sentFromServer <- resp
+	return nil
+}
+func (m *TrainingSetStreamMock) Recv() (*pb.Name, error) {
+	req, more := <-m.recvToServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return req, nil
+}
+
+func (m *TrainingSetStreamMock) SendFromClient(req *pb.Name) error {
+	m.recvToServer <- req
+	return nil
+}
+func (m *TrainingSetStreamMock) RecvToClient() (*pb.TrainingSet, error) {
+	response, more := <-m.sentFromServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return response, nil
+}
+
+func makeStreamMockLabel() *LabelStreamMock {
+	rec := make(chan *pb.Name, 10)
+	sent := make(chan *pb.Label, 10)
+	return &LabelStreamMock{
+		ctx:            context.Background(),
+		recvToServer:   rec,
+		sentFromServer: sent,
+	}
+}
+
+type LabelStreamMock struct {
+	grpc.ServerStream
+	ctx            context.Context
+	recvToServer   chan *pb.Name
+	sentFromServer chan *pb.Label
+}
+
+func (m *LabelStreamMock) Context() context.Context {
+	return m.ctx
+}
+func (m *LabelStreamMock) Send(resp *pb.Label) error {
+	m.sentFromServer <- resp
+	return nil
+}
+func (m *LabelStreamMock) Recv() (*pb.Name, error) {
+	req, more := <-m.recvToServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return req, nil
+}
+
+func (m *LabelStreamMock) SendFromClient(req *pb.Name) error {
+	m.recvToServer <- req
+	return nil
+}
+func (m *LabelStreamMock) RecvToClient() (*pb.Label, error) {
+	response, more := <-m.sentFromServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return response, nil
+}
+
+func makeStreamMockSource() *SourceStreamMock {
+	rec := make(chan *pb.Name, 10)
+	sent := make(chan *pb.Source, 10)
+	return &SourceStreamMock{
+		ctx:            context.Background(),
+		recvToServer:   rec,
+		sentFromServer: sent,
+	}
+}
+
+type SourceStreamMock struct {
+	grpc.ServerStream
+	ctx            context.Context
+	recvToServer   chan *pb.Name
+	sentFromServer chan *pb.Source
+}
+
+func (m *SourceStreamMock) Context() context.Context {
+	return m.ctx
+}
+func (m *SourceStreamMock) Send(resp *pb.Source) error {
+	m.sentFromServer <- resp
+	return nil
+}
+func (m *SourceStreamMock) Recv() (*pb.Name, error) {
+	req, more := <-m.recvToServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return req, nil
+}
+
+func (m *SourceStreamMock) SendFromClient(req *pb.Name) error {
+	m.recvToServer <- req
+	return nil
+}
+func (m *SourceStreamMock) RecvToClient() (*pb.Source, error) {
+	response, more := <-m.sentFromServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return response, nil
+}
+
+func TestCreateMetadataserverMethods(t *testing.T) {
+	sugar := zap.NewExample().Sugar()
+	server := Config{
+		Logger: sugar,
+	}
+	metadataServer, err := NewMetadataServer(&server)
```

Comment:
Can you break all these up into separate tests with cleaner names?

Reply:
@saadhvi27 checkout https://github.com/featureform/embeddinghub/blob/main/ff/serving/provider/offline_test.go as an example.
You can make subtests for each individual test and pass the metadataServer to each of them so you can use dependencies from previous tests.